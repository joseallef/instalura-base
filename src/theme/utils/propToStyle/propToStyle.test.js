/* eslint-disable no-console */
import { propToStyle } from './index';

describe('propToStyle()', () => {
  describe('whe receives an simple argument', () => {
    test('and it is a string', () => {
      const propToStyleResult = propToStyle('textAlign');
      // <Text textAlign="center" />
      const componentProps = { textAlign: 'center' }; // tring
      console.log('propToStyleResult', propToStyleResult(componentProps));
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toEqual({ textAlign: 'center' });
    });
    test('and it is a numbe', () => {
      const propToStyleResult = propToStyle('flex');
      // <Text flex={1} />
      const componentProps = { flex: 1 }; // number
      console.log('propToStyleResult', propToStyleResult(componentProps));
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toEqual({ flex: 1 });
    });
  });
  describe('whe receives an argument with breakpoints', () => {
    test('renders only one breackpoits resolution', () => {
      const propToStyleResult = propToStyle('textAlign');
      // <Text textAlign="center" />
      const componentProps = { textAlign: { xs: 'center' } }; // tring
      console.log('propToStyleResult', propToStyleResult(componentProps));
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toMatchSnapshot();
    });
    test('renders two on more breackpoit resolutions', () => {
      const propToStyleResult = propToStyle('textAlign');
      // <Text textAlign="center" />
      const componentProps = { textAlign: { xs: 'center', md: 'right' } }; // tring
      console.log('propToStyleResult', propToStyleResult(componentProps));
      const styleResult = propToStyleResult(componentProps);
      expect(styleResult).toMatchSnapshot();
    });
  });
  // eslint-disable-next-line jest/no-commented-out-tests
  // test('...', () => {
  //   const propToStyleResult = propToStyle('textAlign');

  //   // <Text textAlign="center" />
  //   const componentProps = { textAlign: 'center' }; // Number || string
  //   console.log('propToStyleResult', propToStyleResult(componentProps));
  //   const styleResult = propToStyleResult(componentProps);

  //   expect(styleResult).toEqual({ textAlign: 'center' });
  // });
});
