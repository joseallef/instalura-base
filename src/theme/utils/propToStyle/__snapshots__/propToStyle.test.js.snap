// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`propToStyle() whe receives an argument with breakpoints renders only one breackpoits resolution 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
]
`;

exports[`propToStyle() whe receives an argument with breakpoints renders two on more breackpoit resolutions 1`] = `
Array [
  Array [
    "@media screen and (min-width:",
    "0",
    "px){",
    "text-align: center;",
    "}",
  ],
  Array [
    "@media screen and (min-width:",
    "768",
    "px){",
    "text-align: right;",
    "}",
  ],
]
`;
